core/src/main/scala/com/banno/kafka/consumer/ConsumerApi.scala

```diff
@@ -86,141 +87,223 @@ trait ConsumerApi[F[_], K, V] {
 
 object ConsumerApi {
 
-  def createConsumer[F[_]: Sync, K, V](configs: (String, AnyRef)*): F[KafkaConsumer[K, V]] =
+  def createKafkaConsumer[F[_]: Sync, K, V](configs: (String, AnyRef)*): F[KafkaConsumer[K, V]] =
     Sync[F].delay(new KafkaConsumer[K, V](configs.toMap.asJava))
 
-  def createConsumer[F[_]: Sync, K, V](
+  def createKafkaConsumer[F[_]: Sync, K, V](
       keyDeserializer: Deserializer[K],
       valueDeserializer: Deserializer[V],
       configs: (String, AnyRef)*
   ): F[KafkaConsumer[K, V]] =
     Sync[F].delay(new KafkaConsumer[K, V](configs.toMap.asJava, keyDeserializer, valueDeserializer))
 
-  def resourceKafkaConsumer[F[_]: Sync, K, V](
+  object BlockingContext {
+
+    def threadFactory: ThreadFactory = new ThreadFactory {
+      val factory = Executors.defaultThreadFactory()
+      def newThread(r: Runnable): Thread = {
+        val t = factory.newThread(r)
```

Comment:
Do we need to worry about this thread leaking. I see default exposes it as a Effect without the resource, in which case this thread would block the jvm from shutting down? 

Should this be a daemon thread? Should the one that is not in resource set this to be a daemon thread?